### PR TITLE
Merge fix from master to develop branch

### PR DESCRIPTION
**Pull request by yamingk**

### CODE REVIEW

File: .jenkins/Dockerfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ LABEL description="Automated HomeStore compilation"
 WORKDIR /output
 
 RUN set -eux; \
+    sed -i 's/master/latest/g' /etc/apt/sources.list; \
```
Comment (hkadayam) [inline .jenkins/Dockerfile:8]: Is this change needed in develop?

File: src/tests/test_meta_blk_mgr.cpp [hunk 2]
```diff
@@ -134,7 +137,7 @@ class VMetaBlkMgrTest : public ::testing::Test {
         return sec.count();
     }
 
-    [[nodiscard]] bool keep_running() {
+        [[nodiscard]] bool keep_running() {
```
Comment (hkadayam) [inline src/tests/test_meta_blk_mgr.cpp:140]: I think there is some misalignment in spaces. Existing change is correct. Perhaps format_on_save not enabled?

File: .jenkins/Dockerfile (hunk 1 above)
Comment (d2chau) [reply .jenkins/Dockerfile:8]: I think so, otherwise when building you can still run into the issue when running apt-get update

File: src/tests/test_meta_blk_mgr.cpp [hunk 3]
```diff
@@ -322,12 +327,12 @@ class VMetaBlkMgrTest : public ::testing::Test {
...
     }
 
-    void do_sb_update(const bool aligned_buf_size) {
+    void do_sb_update(const bool aligned_buf_size, uint64_t size_to_update = 0) {
         ++m_update_cnt;
         uint8_t* buf{nullptr};
         auto overflow = do_overflow();
         if (!aligned_buf_size) { overflow = true; } // for unaligned buf size, let's generate overflow buf size;
-        auto sz_to_wrt = rand_size(overflow, aligned_buf_size);
+        auto sz_to_wrt{size_to_update ? size_to_update : rand_size(overflow, aligned_buf_size)};
```
Comment (hkadayam) [inline src/tests/test_meta_blk_mgr.cpp:335]: I would prefer consistency with develop branch and do `=` for assignments and use `{}` only for object building.

File: src/tests/test_meta_blk_mgr.cpp [hunk 4]
```diff
@@ -415,6 +420,31 @@ class VMetaBlkMgrTest : public ::testing::Test {
...
+    //
+    void write_compression_backoff() {
+        HS_SETTINGS_FACTORY().modifiable_settings([](auto& s) {
+            s.metablk.compress_ratio_limit = 100; // this will allow every compression attempt;
+            HS_SETTINGS_FACTORY().save();
+        });
+
+        LOGINFO("compression ratio limit changed to: {}", m_mbm->get_compress_ratio_limit());
+
+        [[maybe_unused]] const auto write_result = do_sb_write(true /* do_overflow */, 15 * Mi);
```
Comment (hkadayam) [inline src/tests/test_meta_blk_mgr.cpp:435]: we removed the nodiscard from all apis, so this annoying maybe_unused assignment can be skipped and just do ```do_sb_write(...)```

File: src/tests/test_meta_blk_mgr.cpp (hunk 2 above)
Comment (yamingk) [reply src/tests/test_meta_blk_mgr.cpp:140]: my clang format plugin seems to be always get fooled with this keyword "[[nodiscard]]", let me fix it.

File: src/tests/test_meta_blk_mgr.cpp (hunk 3 above)
Comment (yamingk) [reply src/tests/test_meta_blk_mgr.cpp:335]: will do

File: src/tests/test_meta_blk_mgr.cpp (hunk 4 above)
Comment (yamingk) [reply src/tests/test_meta_blk_mgr.cpp:435]: hmm... it yields build failure for me in master, maybe it is not the case in develop branch.

File: .jenkins/Dockerfile (hunk 1 above)
Comment (yamingk) [reply .jenkins/Dockerfile:8]: every branch is failing without this line in image state including AM

File: src/tests/test_meta_blk_mgr.cpp (hunk 4 above)
Comment (yamingk) [reply src/tests/test_meta_blk_mgr.cpp]: /tmp/source/yk_develop/src/tests/test_meta_blk_mgr.cpp:435:20: error: ignoring return value of âuint64_t VMetaBlkMgrTest::do_sb_write(bool, size_t)â, declared with attribute ânodiscardâ [-Werror=unused-result]
  435 |         do_sb_write(true /* do_overflow */, 15 * Mi);
      |         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Review by szmyd — DISMISSED** [review]

**Review by d2chau — APPROVED** [review]
